core-bundle/src/Twig/Global/ContaoVariable.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\Twig\Global;
+
+use Contao\CoreBundle\Csrf\ContaoCsrfTokenManager;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ContaoVariable
+{
+    public function __construct(
+        private readonly ContaoFramework $framework,
+        private readonly RequestStack $requestStack,
+        private readonly TokenChecker $tokenChecker,
+        private readonly ContaoCsrfTokenManager $tokenManager,
+    ) {
+    }
+
+    public function getPage(): PageModel|null
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (!$request || !$request->attributes->has('pageModel')) {
+            return null;
+        }
+
+        $pageModel = $request->attributes->get('pageModel');
+
+        if ($pageModel instanceof PageModel) {
+            return $pageModel;
+        }
+
+        if (
+            isset($GLOBALS['objPage'])
+            && $GLOBALS['objPage'] instanceof PageModel
+            && (int) $GLOBALS['objPage']->id === (int) $pageModel
+        ) {
+            return $GLOBALS['objPage'];
+        }
```

Comment:
This is legacy code from Contao 4.13. In Contao 5, the request always contains a page mode, doesn‘t it?

Reply:
fixed in a0245b588f67b9227d898e81b36b7f3c4848b40d